nodejs/week3/app.js

```diff
@@ -0,0 +1,65 @@
+import knex from "knex";
+import dotenv from "dotenv";
+dotenv.config();
+import express from "express";
+
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  const allowedSortColumns = [
+    "id",
+    "first_name",
+    "last_name",
+    "email",
+    "phone",
+  ];
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString().toLowerCase();
+    const [column, direction] = orderBy.split(" ");
+    const sortDirection = direction ? direction : "asc";
+
+    if (allowedSortColumns.includes(column) && ["asc", "desc"].includes(sortDirection)) {
```

Comment:
Here you are mixing two approaches and consistency is very important. I would pull out the array of asc and dec to a variable next to allowedSortColumns so its easy to see what the valid values are.